File: backend/TimeTracker.Api/Models/TimeRecord.cs
namespace TimeTracker.Api.Models;

public class TimeRecord
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public DateTimeOffset StartTime { get; set; }
    public DateTimeOffset EndTime { get; set; }
    public string Category { get; set; } = string.Empty;
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? UpdatedAt { get; set; }
}

File: frontend/src/types/timeRecord.ts
export interface TimeRecord {
  id: string;
  startTime: string;
  endTime: string | null;
  category: string;
  notes: string | null;
  createdAt: string;
  updatedAt: string | null;
}

export interface TimeRecordInput {
  startTime: string;
  endTime: string | null;
  category: string;
  notes: string | null;
}

export interface CategoryStat {
  category: string;
  totalHours: number;
  recordCount: number;
}

export interface StatsResponse {
  from: string | null;
  to: string | null;
  totalHours: number;
  byCategory: CategoryStat[];
}
